sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ExecutePlanResponseReattachableIterator.scala

```diff
@@ -107,7 +107,7 @@ class ExecutePlanResponseReattachableIterator(
 
   // Server side session ID, used to detect if the server side session changed. This is set upon
   // receiving the first response from the server.
-  private var serverSideSessionId: Option[String] = None
+  private[connect] var serverSideSessionId: Option[String] = None
```

Comment:
Could we create a test without expanding the scope here? Since it's a `var`, I am a bit uncomfortable having it exposed and open to potential modification

Reply:
Understood. The scala changes are optional, so I'll just remove them.

Reply:
are they optional because we already handle this on the Scala side?

Reply:
Yes, there is no problem in the scala code.